docs/going-into-production.rst

```diff
@@ -0,0 +1,175 @@
+.. _going-into-production:
+
+=====================
+Going into production
+=====================
+
+Running CrateDB in different environments requires different approaches. This
+document outlines the basics you need to consider when going into production.
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+
+.. _prod-auto-boot:
+
+Manual bootstrapping
+====================
+
+:ref:`Single host auto-bootstrapping <auto-bootstrapping>` is useful for
+development environments. However, auto-bootstrapping is typically too slow for
+production use. Additionally, for improved performance and resiliency,
+production CrateDB clusters should be run with one node per host machine. If
+you use multiple hosts, you must configure your cluster with :ref:`manual
+bootstrapping <manual-bootstrapping>`.
+
+
+.. _prod-config:
+
+Additional considerations
+=========================
+
+The following setting can be changed in your `configuration`_ file or set via
+command-line arguments. Additionally, you can change some settings at
+`runtime`_.
+
+
+.. _prod-config-path:
+
+Paths
+-----
+
+In a production environment, you should change the default locations of the
+data and log folders to avoid accidental deletion.
+
+.. SEEALSO::
+
+    `Path settings`_
+
+
+.. _prod-config-cluster-name:
+
+Cluster name
+------------
+
+The `cluster.name`_ setting allows you to create multiple separate clusters. A
+node will refuse to join a cluster if the respective cluster names do not
+match.
+
+If you are running multiple clusters on the same network, you should set unique
+cluster names.
+
+.. SEEALSO::
+
+    :ref:`Cluster names for multi-node setups <multi-node-cluster-name>`
+
+
+.. _prod-config-node-name:
+
+Node name
+---------
+
+By default, CrateDB sets the node name for you. However, if you configure the
+node names explicitly, you can specify a list of master-eligible nodes
+up-front.
+
+You can configure a custom node name using the `node.name`_ setting.
+
+.. SEEALSO::
+
+    :ref:`Node names for multi-node setups <multi-node-node-name>`
+
+
+.. _prod-config-network-host:
+
+Network host
+------------
+
+By default, CrateDB binds to the loopback address (i.e., `localhost`_). It
+listens on port 4200-4300 for HTTP traffic and port 4300-4400 for node-to-node
+communication. Because CrateDB uses a port range, if one port is busy, it will
+automatically try the next port.
+
+When using :ref:`multiple hosts <multi_node_setup>`, nodes must bind to a
+non-loopback address.
+
+You can bind to a non-loopback address with the `network.host`_ setting.
+
+.. SEEALSO::
+
+    `Host settings`_
+
+.. CAUTION::
+
+      Never expose an unprotected CrateDB node to the public internet
```

Comment:
Would it make sense to also mention here that bootstrap checks are enforced once CrateDB is bound to a non-loopback address?

Reply:
I think yes.  I wasn't sure whether it was the case!

Reply:
done